### DISCUSSION TITLE
Difference between "change" and "end" in VSlider

### DISCUSSION THREAD

**Discussion by juanprietob** [Kitware/trame#490] [Q&A] [answered]
I have a series of VSlider defined as:
```
vuetify.VSlider(
    v_model=("tx", 0), 
    min=-1.0, max=1, step=0.01,
    hide_details=True, dense=True,
    change=OnTransformSliderChange, # <- the issue is that the change event is only executed after the slider is released
    # end=OnTransformSliderChange # The "change" slot behaves the same as the "end" slot
)
```

The sliders control the translation and rotation of an object. I want the object to update as I move the sliders. I have the following update function:

```
def OnTransformSliderChange():
    translation = np.array([state.tx, state.ty, state.tz])
    rotation_angles = np.array([state.rx, state.ry, state.rz])
    vc.OnTransformSliderChange(translation, rotation_angles)
    ctrl.view_update()
```

Why is the behavior similar to the "end" event? 
Why is the "change" event only triggered when the slider is released?

**Comment by jourdain** [comment, marked as answer]
That is what is described in the vuetify documentation for that component. 

If you want constant update you need to use `input=OnTransformSliderChange`. But I would not recommend doing that.

Instead simply listen on "tx" change. That will happen continuously while not over subscribing the server.

```python
@state.change("tx")
def OnTransformSliderChange(tx, **kwargs):
    ...
```
